### PR TITLE
CDAP-14738 ablity to use py4j library to run python scripts. Mutual authentication

### PR DESCRIPTION
**Pull request by aonischuk**
In case jobs from multiple users running on the same machine. By MITM attack user would be able to access JVM and execute anything there unless we secure the connection.

The connection is secured via TLS.
Java side generates the keystore.

### CODE REVIEW

File: src/main/java/co/cask/hydrator/python/transform/KeyStores.java [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+      keyStore.load(null, password.toCharArray());
+      keyStore.setKeyEntry(CERT_ALIAS, pair.getPrivate(), password.toCharArray(),
+                           new java.security.cert.Certificate[]{cert});
+      return keyStore;
+    } catch (Exception e) {
+      throw new RuntimeException("SSL is enabled but a key store file could not be created. A keystore is required " +
+                                   "for SSL to be used.", e);
+    }
+  }
+  /**
```
Comment (albertshau) [inline src/main/java/co/cask/hydrator/python/transform/KeyStores.java:125]: nit: add a newline between previous method and this javadoc

File: src/main/java/co/cask/hydrator/python/transform/Py4jPythonExecutor.java [hunk 2]
```diff
@@ -131,9 +166,30 @@ public void initialize(ScriptContext scriptContext) throws IOException, Interrup
...
+
+    TrustManagerFactory tmf = TrustManagerFactory.getInstance("SunX509");
+    tmf.init(keyStore);
+
+    // setup the HTTPS context and parameters
+    sslContext.init(kmf.getKeyManagers(), tmf.getTrustManagers(), null);
+
+
+    gatewayServer = new GatewayServer((Object) null, 0,
+                                             InetAddress.getByName("localhost"), GatewayServer.DEFAULT_CONNECT_TIMEOUT,
```
Comment (albertshau) [inline src/main/java/co/cask/hydrator/python/transform/Py4jPythonExecutor.java]: nit: the indentation here looks off, in the middle of the line.

**Review by albertshau — COMMENTED** [review]
minor nits, otherwise lgtm. I'm not familiar with the cert logic, but will just assume it's been tested.

File: src/main/java/co/cask/hydrator/python/transform/KeyStores.java (hunk 1 above)
Comment (aonischuk) [reply src/main/java/co/cask/hydrator/python/transform/KeyStores.java:125]: fixed

File: src/main/java/co/cask/hydrator/python/transform/Py4jPythonExecutor.java (hunk 2 above)
Comment (aonischuk) [reply src/main/java/co/cask/hydrator/python/transform/Py4jPythonExecutor.java]: fixed

**Comment by aonischuk** [comment]
@albertshau it was surely :)
Fixed your review comments.
Can you please merge?

**Review by albertshau — APPROVED** [review]
lgtm

**Pull request merged**